### PR TITLE
Output dynamic component type

### PR DESCRIPTION
**Pull request by ostrgard**
_Picking up on the stale PR: https://github.com/strapi/gatsby-source-strapi/pull/99_

Currently the dynamic component type is being stripped out.

CC @alexandrebodin

### CODE REVIEW

File: src/fetch.js
Comment (alexandrebodin) [inline src/fetch.js:34]: Just wondering but an't we just leave the __component field ?

Comment (ostrgard) [reply src/fetch.js:34]: Hmm the following if clause makes me think we're trying to stear clear of underscore-prepended fields, but I think that should be your call :)

Comment (alexandrebodin) [reply src/fetch.js:34]: Yeah tbh I would clear this as I don't see the point of all this... I'm just wondering if this was done to avoid bugs with gatsby (or graphql) not liking underscores as first character... Would you mind trying it out ?

Comment (ostrgard) [reply src/fetch.js]: Just tried it and yeah, it will output, but is of course converted to `_component` due to the following if clause. I feel that's a pretty weird side-effect.

This https://github.com/strapi/gatsby-source-strapi/commit/3ac3d8b171f3fc32f3c3d277b0b428132f2d7f8d is all the context I could find on this logic. I guess this is also mongoose related, and should be handled with more specificity I think. See new commit.

@pierreburgy , is this something you can chime in on? Thanks :)

Comment (ostrgard) [reply src/fetch.js]: Woah, now that it's not removing the first underscore, it seems that Gatsby renames `__component` to `_xcomponent`. Haven't dug too deep into this, but it seems that Gatsby reserves `__`. Also a pretty weird side-effect. I'd say going with a more specifically named key like my initial commit would actually be appropriate in this case.

Comment (alexandrebodin) [reply src/fetch.js]: OK maybe the __ keys other than __v could be prefixed with strapi so we know they are specific to strapi

Comment (ostrgard) [reply src/fetch.js]: Good call. Check my newest commit.

**Comment by YannickMol** [comment]
Hi guys, any update on this? Would love to use the dynamic zones in Gatsby!

**Comment by ostrgard** [comment]
Awaiting final signoff from @alexandrebodin :)

**Review by alexandrebodin — APPROVED** [review]

**Comment by alexandrebodin** [comment]
Thanks for the contribution 💯

**Comment by lucastobrazil** [comment]
Does this have any documentation? i am trying to pick it up in Gatsby GraphQL but i cannot see the field

**Comment by alexandrebodin** [comment]
This was not released yet. you can install it using the github url if you really need to with npm.

**Comment by lucastobrazil** [comment]
@alexandrebodin just did it - works! excited to have this feature. Are there any PRs/ efforts going into docs for this plugin? Happy to contribute!

**Comment by alexandrebodin** [comment]
Sadly we are still lacking devs in the team to be able to maintain this as good as we would like to. We sure are open to get contributors if you are interested drop by the slack to discuss with us ;)

**Comment by lucastobrazil** [comment]
@alexandrebodin awesome! i am in the Strapi slack, (strapi.slack.com), but haven't heard from anyone about Gatsby. Is there another space? :)

**Comment by alexandrebodin** [comment]
Nope that's the one :) just leave a message to @derrickmehaffy  or @sam-pires to talk about contributions ;)

**Comment by sam-pires** [comment]
Hello @lucastobrazil 
Let us know if you have questions and/or suggestions about contributing  :)

**Pull request merged**
